core/chat/events.go

```diff
@@ -0,0 +1,98 @@
+package chat
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/owncast/owncast/core/chat/events"
+	"github.com/owncast/owncast/core/data"
+	"github.com/owncast/owncast/core/user"
+	"github.com/owncast/owncast/core/webhooks"
+	log "github.com/sirupsen/logrus"
+)
+
+func (s *ChatServer) userNameChanged(eventData chatClientEvent) {
+	var receivedEvent events.NameChangeEvent
+	if err := json.Unmarshal(eventData.data, &receivedEvent); err != nil {
+		log.Errorln("error unmarshalling to NameChangeEvent", err)
+		return
+	}
+
+	proposedUsername := receivedEvent.NewName
+	blocklist := data.GetUsernameBlocklist()
+
+	for _, blockedName := range blocklist {
+		normalizedName := strings.TrimSpace(blockedName)
+		normalizedName = strings.ToLower(normalizedName)
+		if strings.Contains(normalizedName, proposedUsername) {
+			// Denied.
+			log.Debugln(eventData.client.User.DisplayName, "blocked from changing name to", proposedUsername, "due to blocked name", normalizedName)
+			message := fmt.Sprintf("You cannot change your name to **%s**.", proposedUsername)
+			s.sendActionToClient(eventData.client, message)
+			return
+		}
+	}
+
+	savedUser := user.GetUserByToken(eventData.client.accessToken)
+	oldName := savedUser.DisplayName
+
+	// Save the new name
+	user.ChangeUsername(eventData.client.User.Id, receivedEvent.NewName)
+
+	// Send chat event letting everyone about about the name change
+	savedUser.DisplayName = receivedEvent.NewName
+
+	// Update the connected clients associated user with the new name
+	eventData.client.User = savedUser
```

Comment:
Swap, so the comments stay correct:

```suggestion
	// Update the connected clients associated user with the new name
	eventData.client.User = savedUser

	// Send chat event letting everyone about about the name change
	savedUser.DisplayName = receivedEvent.NewName
```